### PR TITLE
CONTRIBUTING.md: describe coding style more accurately

### PR DESCRIPTION
**Pull request by johanmalm**
Add a few more bits :smile: including

- Clarify the coding style relative to Linux kernel and Devault rules
- Add g_pattern_match_simple() example
- Add namespace advice and preference

### CODE REVIEW

**Comment by Consolatis** [comment]
LGTM

**Comment by johanmalm** [comment]
v2: 
- Add TOC
- Elaborate on header documentation

Will add something on adding items to end of list, but let's conclude that discussion separately.

Maybe we ought to add some stuff add address "I'd like to contribute, but don't know C". There are a few OO design patterns that we use which could do with being explained. Also, the use of the linked lists could do with a simple example.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -75,19 +90,80 @@ to open an issue so we can add you to this list.
...
+Put `*` with the identifier when defining pointers, for example `char *foo`
+
+Spaces are placed around binary operators but not unary, like this:
+
+`sizeof(*foo)` is preferred over `sizeof(struct foo)`
+
+Use `if (!p)` instead of `if (p == 0)` or `if (p == NULL)`
+
+```
+int x = y * (2 + z);
```
Comment (Consolatis) [inline CONTRIBUTING.md:125]: This code block belongs to "Spaces are placed around binary operators [..]"

Comment (johanmalm) [reply CONTRIBUTING.md:121]: Thanks. Fixed.

**Comment by johanmalm** [comment]
v3
- Added `wl_list_append()` to API

Will merge now.

**Pull request merged**